core-rust/node-common/src/scheduler.rs

```diff
@@ -62,57 +62,227 @@
  * permissions under this License.
  */
 
-use std::ops::Div;
+use std::sync::Arc;
 
-use clokwerk::{Interval, ScheduleHandle};
+use core::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
 use std::time::Duration;
 
+use tokio::runtime::Runtime;
+use tokio_util::sync::{CancellationToken, WaitForCancellationFuture};
+
+use crate::locks::{LockFactory, Mutex};
+
+pub trait CancellableToken {
+    type VoidFuture<'a>: Future<Output = ()>
+    where
+        Self: 'a;
+
+    /// Be aware that cancellation is not an atomic operation. It is possible
+    /// for another thread running in parallel with a call to `cancel` to first
+    /// receive `true` from `is_cancelled` on one child node, and then receive
+    /// `false` from `is_cancelled` on another child node. However, once the
+    /// call to `cancel` returns, all child nodes have been fully cancelled.
+    fn cancel(&self);
```

Comment:
Pretty please, let's leave just this single method in the public interface of the scheduler facade v1 🙏 .